ui/apps/dashboard/e2e/member clusters/test-utils.ts

```diff
@@ -0,0 +1,228 @@
+
+/**
+ * Creates a configured Kubernetes API client for Karmada
+ * @returns Kubernetes CustomObjectsApi client
+ */
+function createKarmadaApiClient(): k8s.CustomObjectsApi {
+    const kc = new k8s.KubeConfig();
+
+    // Try to use existing kubeconfig first (for CI)
+    if (process.env.KUBECONFIG) {
+        try {
+            kc.loadFromFile(process.env.KUBECONFIG);
+            // Set context to karmada-apiserver if available
+            const contexts = kc.getContexts();
+            const karmadaContext = contexts.find(c => c.name === 'karmada-apiserver');
+            if (karmadaContext) {
+                kc.setCurrentContext('karmada-apiserver');
+            }
+            return kc.makeApiClient(k8s.CustomObjectsApi);
+        } catch (error) {
+            console.error('Failed to load kubeconfig:', error);
+        }
+    }
+
+    // Fallback to custom config for local development
+    const kubeConfigYaml = `
+apiVersion: v1
+kind: Config
+clusters:
+- cluster:
+    insecure-skip-tls-verify: true
+    server: ${KARMADA_API_SERVER}
+  name: karmada-apiserver
+contexts:
+- context:
+    cluster: karmada-apiserver
+    user: karmada-dashboard
+  name: default
+current-context: default
+users:
+- name: karmada-dashboard
+  user:
+    token: ${token}
+`;
+
+    kc.loadFromString(kubeConfigYaml);
+    return kc.makeApiClient(k8s.CustomObjectsApi);
+}
+
+/**
+ * Setup dashboard authentication and navigate to dashboard page
+ */
+export async function setupDashboardAuthentication(page: Page) {
+    await page.goto(`${baseURL}/login`, { waitUntil: 'networkidle' });
+    await page.evaluate((t) => localStorage.setItem('token', t), token);
+    await page.goto(`${baseURL}${basePath}`, { waitUntil: 'networkidle' });
+    await page.evaluate((t) => localStorage.setItem('token', t), token);
+    await page.reload({ waitUntil: 'networkidle' });
+    await page.waitForSelector('text=Overview', { timeout: 30000 });
+}
+
+/**
+ * Generate test Member Cluster kubeconfig YAML with timestamp
+ * Note: The Member Cluster creation page expects kubeconfig format, not Cluster resource format
+ */
+export function generateTestMemberClusterYaml() {
+    const timestamp = Date.now();
+    return `apiVersion: v1
+kind: Config
+clusters:
+- cluster:
+    insecure-skip-tls-verify: true
+    server: https://test-cluster-${timestamp}.example.com:6443
+  name: test-membercluster-${timestamp}
+contexts:
+- context:
+    cluster: test-membercluster-${timestamp}
+    user: test-user-${timestamp}
+  name: test-context-${timestamp}
+current-context: test-context-${timestamp}
+users:
+- name: test-user-${timestamp}
+  user:
+    token: test-token-${timestamp}`;
+}
+
+/**
+ * Generate test Member Cluster resource YAML with timestamp in Cluster resource format
+ * Note: This generates a Cluster resource that can be created via Karmada API
+ */
+export function generateTestMemberClusterResourceYaml() {
```

Comment:
I feel that this method can be abstracted to be more general, independent of specific resource types.

Reply:
ok